### DISCUSSION TITLE
Homebrew installation fails with Error: Failed to install ruby 2.6.8!

### DISCUSSION THREAD

**Discussion by ncpa0cpl** [Homebrew/discussions#3183] [Linux] [answered]
When installing homebrew via the official script it fails when installing Ruby.

```
==> Checking for `sudo` access (which may request your password)...
==> This script will install:
/home/linuxbrew/.linuxbrew/bin/brew
/home/linuxbrew/.linuxbrew/share/doc/homebrew
/home/linuxbrew/.linuxbrew/share/man/man1/brew.1
/home/linuxbrew/.linuxbrew/share/zsh/site-functions/_brew
/home/linuxbrew/.linuxbrew/etc/bash_completion.d/brew
/home/linuxbrew/.linuxbrew/Homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /bin/chown -R owner:owner /home/linuxbrew/.linuxbrew/Homebrew
==> Downloading and installing Homebrew...
HEAD is now at 97b016254 Merge pull request #13138 from danielnachun/patch_prefix3
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:cd7fffb18ef9338baa670fc5e8fce99b0e3cc0f0fd7627bcbb56f3c8d54161d4
Already downloaded: /home/owner/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
==> Pouring portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
Error: Failed to install ruby 2.6.8!
Error: Failed to upgrade Homebrew Portable Ruby!
If there's no Homebrew Portable Ruby available for your processor:
- install Ruby 2.6.8 with your system package manager (or rbenv/ruby-build)
- make it first in your PATH
- try again

Failed during: /home/linuxbrew/.linuxbrew/bin/brew update --force --quiet`
```
The error message is not helpful at all, it does not tell me what exactly went wrong so I don't have any idea on how to approach this.

I've tried to install ruby via `rbenv` but it did not change anything, here's my current ruby version:

```
$ ruby -v
ruby 2.6.8p205 (2021-07-07 revision 67951) [x86_64-linux]
```

My operating system is Manjaro Linux with the kernel `5.10.105-1-MANJARO`

Any help would be appreciated.

**Comment by SMillerDev** [comment]
Can you run `ruby -v` and the installer and then paste all output here?

**Reply by ncpa0cpl** [reply]
I mean, I just did, but sure I can do it again:
```
$ ruby -v
ruby 2.6.8p205 (2021-07-07 revision 67951) [x86_64-linux]
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
==> Checking for `sudo` access (which may request your password)...
==> This script will install:
/home/linuxbrew/.linuxbrew/bin/brew
/home/linuxbrew/.linuxbrew/share/doc/homebrew
/home/linuxbrew/.linuxbrew/share/man/man1/brew.1
/home/linuxbrew/.linuxbrew/share/zsh/site-functions/_brew
/home/linuxbrew/.linuxbrew/etc/bash_completion.d/brew
/home/linuxbrew/.linuxbrew/Homebrew

Press RETURN/ENTER to continue or any other key to abort:
Place your finger on the fingerprint reader
==> /usr/bin/sudo /bin/chown -R owner:owner /home/linuxbrew/.linuxbrew/Homebrew
==> Downloading and installing Homebrew...
HEAD is now at 97b016254 Merge pull request #13138 from danielnachun/patch_prefix3
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:cd7fffb18ef9338baa670fc5e8fce99b0e3cc0f0fd7627bcbb56f3c8d54161d4
Already downloaded: /home/owner/.cache/Homebrew/portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
==> Pouring portable-ruby-2.6.8.x86_64_linux.bottle.tar.gz
Error: Failed to install ruby 2.6.8!
Error: Failed to upgrade Homebrew Portable Ruby!
If there's no Homebrew Portable Ruby available for your processor:
- install Ruby 2.6.8 with your system package manager (or rbenv/ruby-build)
- make it first in your PATH
- try again

Failed during: /home/linuxbrew/.linuxbrew/bin/brew update --force --quiet
```

**Reply by Bo98** [reply]
Does `/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/current/bin/ruby --version` output anything?

**Comment by teslamint** [comment, marked as answer]
It appears Manjaro deprecates `libcrypt.so.1` so portable ruby 2.6.8 complains about it.
Install `libxcrypt-compat` will resolve the issue.
Some backgrounds: https://bbs.archlinux.org/viewtopic.php?id=274160

**Reply by Bo98** [reply]
We're working on deprecating `libcrypt.so.1` ourselves, but this is indeed the solution for now.

**Reply by ryderstorm** [reply]
Thank you so much for this! i'd been trying to work around this issue for a week or more. Your solution worked perfectly.

**Reply by ncpa0cpl** [reply]
This fixed the issue, thank you for help

**Reply by olivertzeng** [reply]
hohohohohohh my gooooddddddd it works!!!!!

**Reply by Yasir-Khan22** [reply]
Hi! Buddy, what's the issue now.
![erro1](https://user-images.githubusercontent.com/63096083/187065522-990c8550-f19b-4f3b-aa0d-afc0ed60f375.PNG)
However, I've installed ruby latest version.
![ruby-version](https://user-images.githubusercontent.com/63096083/187065541-a9afa71b-300d-4a6e-9efb-1375202cfbf8.PNG)

I'm using ubuntu on windows as WSL.

**Reply by Bo98** [reply]
@Yasir-Khan22 Homebrew's Portable Ruby no longer depends on `libcrypt.so.1` so you likely have a different issue.

**Comment by Feistybeard** [comment]
I had the exact same problem running on Arch with 5.17.1-zen1-1-zen. Installing libxcrypt-compat as suggested by @teslamint solved the issue for me.

**Comment by Feruz00** [comment]
I had a this error. I cannot solve this. But my use Windows 10. I watch course Angela Yu and do what do you. Initially I get this:
`curl: (7) Failed to connect to raw.githubusercontent.com port 443: Connection refused`. I solve this using [this](https://florahuo.medium.com/edit-etc-hosts-file-in-ubuntu-command-line-using-nano-ide-44c3a78e23ca). After I get this error. I didnt know what do after. Please help me.

**Comment by cmallikarjun080** [comment]
Hi i am facing this issue please help me to resolve 

 brew upgrade                                                                                    
==> Downloading https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:1f50bf80583bd436c9542d4fa5ad47df0ef0f0bea22ae710c4f04c42d7560bca
/usr/local/Homebrew/Library/Homebrew/cmd/vendor-install.sh: line 162: /path/to/new/curl: No such file or directory
==> Downloading https://github.com/Homebrew/homebrew-portable-ruby/releases/download/2.6.8_1/portable-ruby-2.6.8_1.el_capitan.bottle.tar.gz
/usr/local/Homebrew/Library/Homebrew/cmd/vendor-install.sh: line 162: /path/to/new/curl: No such file or directory
Error: Failed to download ruby from the following locations:
  - https://ghcr.io/v2/homebrew/portable-ruby/portable-ruby/blobs/sha256:1f50bf80583bd436c9542d4fa5ad47df0ef0f0bea22ae710c4f04c42d7560bca
  - https://github.com/Homebrew/homebrew-portable-ruby/releases/download/2.6.8_1/portable-ruby-2.6.8_1.el_capitan.bottle.tar.gz

Do not file an issue on GitHub about this; you will need to figure out for
yourself what issue with your internet connection restricts your access to
GitHub (used for Homebrew updates and binary packages).
Error: Failed to upgrade Homebrew Portable Ruby!

**Reply by ncpa0cpl** [reply]
This looks like the script cannot resolve the path to the `curl` binary, do you have it installed? I won't be able to help you much further since it is unrelated to the original issue I was having. I'd suggest making a new thread in [Discussions](https://github.com/orgs/Homebrew/discussions/categories/linux), you have a better chance of someone with knowledge on the install script helping you that way.

**Comment by HoppMarc** [comment]
Error: Failed to install Homebrew Portable Ruby and cannot find another Ruby 2.6.8!

I have the same issue on a Xubuntu machine; neither apt nor snap have the libxcrypt.  

ruby -v returns: ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux-gnu]

**Comment by Malik056** [comment]
Installing `curl` using `sudo apt-get install curl` fixing the issue. installing curl using `snap` throw errors.

**Reply by Cemow** [reply]
That saved me thanks.

**Reply by BOVAGE** [reply]
That saved me too thanks.

**Reply by rkuang9** [reply]
That saved me also thanks.
